plugins/mikrotik/mt_system

```diff
@@ -0,0 +1,478 @@
+      print "write_sect_total.value " $2
+      }'
+  done <<< "$data"
+}
+
+function get_write_sect_since_reboot_label {
+  while read -r line; do
+    echo "$line" | awk '/write-sect-since-reboot:/{
+      print "multigraph write_sect_since_reboot_graph_""'"$name"'";
+      print "graph_title Sector writes since reboot " "'"$name"'";
+      print "graph_vlabel count";
+      print "graph_category mikrotik";
+      print "graph_args -l 0";
+      print "write_sect_since_reboot.label write_sect_since_reboot";
+      print "graph_info Total Sector writes since last reboot."
+      }'
+  done <<< "$data"
+}
+function get_write_sect_since_reboot_value {
+  while read -r line; do
+    if [ "$MUNIN_CAP_DIRTYCONFIG" = "0" ] || [ -z "$MUNIN_CAP_DIRTYCONFIG" ]; then
+      echo "$line" | awk '/write-sect-since-reboot:/{
+        print "multigraph write_sect_since_reboot_graph_""'"$name"'";
+      }'
+    fi
+    echo "$line" | awk '/write-sect-since-reboot:/{
+      print "write_sect_since_reboot.value " $2
+      }'
+  done <<< "$data"
+}
+
+function get_temperature_label {
+  while read -r line; do
+    echo "$line" | awk '/temperature/{
+      print "multigraph temperature_graph_""'"$name"'";
+      print "graph_title temperature " "'"$name"'";
+      print "graph_vlabel °C";
+      print "graph_category mikrotik";
+      print "graph_args -l 0";
+      print "temperature.label cpu temperature";
+      print "temperature.warning 75";
+      print "temperature.critical 90"
+      }'
+  done <<< "$data"
+}
+function get_temperature_value {
+  get_ros_version
+  while read -r line; do
+    if [ "$MUNIN_CAP_DIRTYCONFIG" = "0" ] || [ -z "$MUNIN_CAP_DIRTYCONFIG" ]; then
+      echo "$line" | awk '/temperature/{
+        print "multigraph temperature_graph_""'"$name"'";
+        }'
+    fi
+    if [ "$ros_version" == "6" ]; then
+      echo "$line" | awk '/temperature:/{
+        gsub(/C/,"",$2);
+        print "temperature.value " $2
+        }'
+    fi
+    if [ "$ros_version" == "7" ]; then
+      echo "$line" | awk '/cpu-temperature/{
+        print "temperature.value " $3
+        }'
+    fi
+  done <<< "$data"
+}
+
+function get_cpu_label {
+  echo multigraph cpu_load_graph_"$name"
+  echo graph_title cpu load "$name"
+  echo graph_vlabel %
+  echo graph_category mikrotik
+  echo graph_args -l 0 --upper-limit 100
+  echo cpu_total.label total load
+  echo cpu_total.warning 75
+  echo cpu_total.critical 90
+  echo graph_info Total CPU Load and Load, IRQ and Disk per Core.
+
+  while [ "$c" -lt "$anzahl_cpu" ]; do
+    while read -r line; do
+      echo "$line" | grep cpu$c | awk '{
+        print "cpu"'"$c"'"_load.label " "cpu" "'"$c"'" " load"
+        print "cpu"'"$c"'"_irq.label " "cpu" "'"$c"'" " irq"
+        print "cpu"'"$c"'"_disk.label " "cpu" "'"$c"'" " disk"
+        }'
+    done <<< "$data"
+    c=$(( c + 1 ))
+  done
+}
+function get_cpu_value {
+  if [ "$MUNIN_CAP_DIRTYCONFIG" = "0" ] || [ -z "$MUNIN_CAP_DIRTYCONFIG" ]; then
```

Comment:
I thought that was used for getting all values in jus one run, instead of twos?

Reply:
`MUNIN_CAP_DIRTYCONFIG` informs the plugin, that the master tolerates a mixture of data and config in response to a `config` request (see the [guide](http://guide.munin-monitoring.org/en/latest/plugin/protocol-dirtyconfig.html)).
`MUNIN_CAP_MULTIGRAPH` informs the plugin, that the master can deal with the `multigraph` keyword. In this case you may emit nested graphs or simply more than one graph within a single plugin. See the [guide](http://guide.munin-monitoring.org/en/latest/plugin/multigraphing.html) for details.

In this case I *think*, you are referring to *multigraph*. But I have to admit, that I do not fully understand the related comment around line 111.

Reply:
I'll do both, i create multiple graphs with that plugin, and i set dirtyconfig, so that for one munin-"run" only one ssh-connection is made, instead of two for plugin config and after that plugin data.